### PR TITLE
Cassandra 4.0

### PR DESCRIPTION
**Pull request by dhirsch1**

### CODE REVIEW

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-studio.adoc [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+|===
+Field a|Description
+|Keyspace | Apache Cassandra keyspace
+|Username | Apache Cassandra user name
+|Password | Apache Cassandrad password.
+|===
++
+You can reference a configuration file that contains ANT-style property placeholders (recommended), or you can enter your authorization credentials in the global configuration properties. For information about the benefits of using property placeholders and how to configure them, see xref:connectors::introduction/intro-connector-configuration-overview.adoc[Anypoint Connector Configuration].
++
+. On the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.
```
Comment (valean27) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-studio.adoc:106]: Can you please add here a description of the security tab as well?  I think I've put it only on design center doc. Thanks

**Review by valean27 — DISMISSED** [review]

**Review by valean27 — APPROVED** [review]

File: salesforce/10.5/modules/ROOT/pages/index.adoc [hunk 2]
```diff
@@ -35,7 +35,13 @@ If you plan to create the XML for your Mule app, include the correct namespace a
 
 * New users
 +
+<<<<<<< HEAD
```
Comment (valean27) [inline salesforce/10.5/modules/ROOT/pages/index.adoc]: >>>> These are some merging conflicts, please remove the wrong ones.

**Review by valean27 — DISMISSED** [review]

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+
+. Create a new Mule Project in Anypoint Studio.
+. In Studio, click *HTTP* and drag the *Listener* operation to the canvas.
+. On the Listener properties window, click the plus sign (*+*) next to the *Connector configuration* field to add a global element.
+. Accept the defaults.
+. In *Listener* configuration window, set the *Path* field value to `/`.
+
+==== Add a Parse Template Component
+
+Add a *Parse Template* component to create a template for an HTML form. The example uses this form to convey information for creating the table. (*Reviewers, is this text correct?*):
```
Comment (valean27) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc]: yes

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc [hunk 4]
```diff
@@ -0,0 +1,575 @@
...
+
+. From the Mule Palette view, search for `parse` and drag *Core* > *Parse Template* to the right of *Listener*.
+. In the *Location* field, enter `form.html`.
+*Reviewers, where is the information for this template? I assume it is in form.html, but where is the form put?*
```
Comment (valean27) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc]: form.html is located in main/resources of the demo

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc [hunk 5]
```diff
@@ -0,0 +1,575 @@
...
+
+. Create a new Mule Project in Anypoint Studio.
+. In Studio, click *HTTP* and drag the *Listener* operation to the canvas.
+. On the Listener properties window, click the plus sign *+* next to the *Connector configuration* field to add a global element.
+. Accept the defaults.
+. In *Listener* configuration window, set the *Path* field value to `/test/createKeyspace`.
+
+==== Add a Transform Message Component
+
+Add a *Transform Message* component to set Java variables based on the payload received from *HTTP Listener*: *Reviewers, is this description correct*?
```
Comment (valean27) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc]: yes

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc [hunk 6]
```diff
@@ -0,0 +1,575 @@
...
+ "replicationStrategyClass": payload.replicationStrategyClass
+}
+as Object {
+ class: "org.mule.modules.cassandradb.api.CreateKeyspaceInput"
+}
+----
+
+==== Add the Create Keyspace Operation
+
+Add the *Create keyspace* operation to create the new keyspace for the new table: (*Reviewers: please correct and make this text better*)
```
Comment (valean27) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc]: Create a keyspace for the new table to exist in. The tables are created in a keyspace, so you need to create a keyspace first.

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc [hunk 7]
```diff
@@ -0,0 +1,575 @@
...
+
+. Create a new Mule Project in Studio.
+. In Studio, click *HTTP* and drag the *Listener* operation to the canvas.
+. In the *Listener* configuration, click the plus sign *+* next to the *Connector configuration* field to add a global element.
+. Accept the defaults.
+. In the *Listener* configuration, set the *Path* field value to `/test/createTable`.
+
+=== Add a Transform Message Component
+
+Add a *Transform Component* to set Java variables based on the payload received from *HTTP Listener*: ()*Reviewers, is this description correct*)?
```
Comment (valean27) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc]: yes

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc [hunk 8]
```diff
@@ -0,0 +1,575 @@
...
+  "tableName": payload.tableName,
+  "keyspaceName": payload.keyspaceName
+} as Object {
+  class : "org.mule.modules.cassandradb.api.CreateTableInput"
+}
+----
+
+==== Add the Create Table Operation
+
+Add a *Create Table* operation to create a new table based on a POST request that uses the template created in the first flow: (*Reviewers, please make this text more accurate. I am guessing here.*)
```
Comment (valean27) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc]: Not sure what that description is but, we need to add a create table operation in ordeer to create that table on the keyspace that was created before, create table operations takes as parameters a keyspace name, a table name, and a list columns that have a name, a type and a flag if they are primary key or not.  I hope I gave you enough info.

**Review by valean27 — DISMISSED** [review]

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-config-topics.adoc [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+bin/cqlsh localhost -u cassandra -p cassandra
+Connected to Test Cluster at localhost:9042.
+[cqlsh 5.0.1 | Cassandra 3.10 | CQL spec 3.4.4 | Native protocol v4]
+Use HELP for help.
+----
+
+. After installing Apache Cassandra, open the `conf\cassandra.yaml` file for editing.
+. Optionally, change the value of `AllowAllAuthenticator` to `PasswordAuthenticator`.
+. Restart Cassandra.
+. In a separate terminal tab, verify that Cassandra is running:
```
Comment (kevintroller) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-config-topics.adoc]:
```suggestion
. Restart Apache Cassandra.
. In a separate terminal tab, verify that Apache Cassandra is running:
```

Comment (kevintroller) [reply cassandra/4.0/modules/ROOT/pages/cassandra-connector-config-topics.adoc]: Just to make it consistent every time you call Cassandra.

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-config-topics.adoc [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
++
+`bin/nodetool status`
+
+== POJO Configuration
+
+The connector exposes multiple operations in the Apache Cassandra Connector Reference.
+The below operations accept some POJOs as input parameters, which are needed for the DataSense functionality:
```
Comment (kevintroller) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-config-topics.adoc]:
```suggestion
The following operations accept some POJOs as input parameters, which are needed for the DataSense functionality:
```

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-config-topics.adoc [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+`addNewColumn`, `changeColumnType`, `createKeyspace`, `createTable`, `deleteColumnsValue`, `deleteRows`, `dropColumn`, `dropKeyspace`, `dropTable`, `executeCQLQuery`, `getTableNamesFromKeyspace`, `renameColumn`, `select`, and `update`.
```
Comment (kevintroller) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-config-topics.adoc]:
```suggestion
* `addNewColumn` 
* `changeColumnType` 
* `createKeyspace` 
* `createTable` 
* `deleteColumnsValue` 
* `deleteRows` 
* `dropColumn` 
* `dropKeyspace` 
* `dropTable` 
* `executeCQLQuery` 
* `getTableNamesFromKeyspace` 
* `renameColumn` 
* `select` 
* `update`
```

Comment (kevintroller) [reply cassandra/4.0/modules/ROOT/pages/cassandra-connector-config-topics.adoc]: I think it's easier to scan, but feel free to update it or not.

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-design-center.adoc [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+|Field |Description
+|Protocol | HTTP or HTTPS (secure)
+|Host| IP address where the Mule application listens for requests
+|Port| Address where the Mule application listens for requests
+|Base Path| Path where the Mule application listens for requests
+|===
+. Select *+* to add a component.
+. Select the connector as a component.
+. Select the operation.
+. Click the Cassandra connector to configure the host, port, keyspace, and authentication:
```
Comment (kevintroller) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-design-center.adoc]:
```suggestion
. Click Cassandra Connector to configure the host, port, keyspace, and authentication:
```

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc [hunk 13]
```diff
@@ -0,0 +1,575 @@
+= Apache Cassandra Connector 4.0 Examples - Mule 4
+:page-aliases: connectors::cassandra/cassandra-connector-examples.adoc
+
+The following examples show how to configure Apache Cassandra Connector:
+
+* <<create-table,Create a New Apache Cassandra Table>>
++
+Creates a template for an HTML form to use to enter data for the new table, creates a keyspace for the new table, and then creates the table
+* <<manipulate-data,Manipulate Data in an Apache Cassandra Database>>
```
Comment (kevintroller) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc:9]:
```suggestion
* <<create-table,Create a New Apache Cassandra Table>> +
Creates a template for an HTML form to use to enter data for the new table, creates a keyspace for the new table, and then creates the table
* <<manipulate-data,Manipulate Data in an Apache Cassandra Database>>
Manipulate data in an Apache Casandra database by getting table names from a keyspace, performing CRUD operations, executing CQL queries and so on.
```

Comment (kevintroller) [reply cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc:9]: Add an explanation sentence for the second example? Something like the above...

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc [hunk 14]
```diff
@@ -0,0 +1,575 @@
...
+[create-table]
+== Create a New Apache Cassandra Table
+
+This example contains the following flows:
+
+* The first flow configures a template for the HTML input form used to convey data for the new table.
+* The second flow creates the keyspace for the new table.
+* The third flow creates the new table.
+
+This example variables for some field values. You can either:
```
Comment (kevintroller) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc]:
```suggestion
This example uses variables for some field values. You can either:
```

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc [hunk 15]
```diff
@@ -0,0 +1,575 @@
...
+.Flow for configuring an HTML template
+image::cassandra-parse-template.png[Parse Template Flow]
+
+==== Configure HTTP Listener
+
+Configure HTTP Listener to initiate a Mule flow when a call is made to the `/` path on port 8081:
+
+. Create a new Mule Project in Anypoint Studio.
+. In Studio, click *HTTP* and drag the *Listener* operation to the canvas.
+. On the Listener properties window, click the plus sign (*+*) next to the *Connector configuration* field to add a global element.
```
Comment (kevintroller) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc]:
```suggestion
. On the *Listener* properties window, click the plus sign (*+*) next to the *Connector configuration* field to add a global element.
```

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc [hunk 16]
```diff
@@ -0,0 +1,575 @@
...
+}
+----
+
+==== Add the Create Table Operation
+
+Add a *Create Table* operation to create a new table in the keyspace you created in the previous flow:
+
+. From the *Mule Palette* view, select *CassandDB* and drag the *Create Table* operation to the right of the *Transform Message* component.
+. In the *Create Table* configuration, click the plus sign (*+*) next to the *Connector configuration* field.
+. Select global element that you configured for the *Create keyspace* operation.
```
Comment (kevintroller) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc]:
```suggestion
. Select the global element that you configured for the *Create keyspace* operation.
```

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc [hunk 17]
```diff
@@ -0,0 +1,575 @@
...
+. Click *OK*.
+. Click *Test Connection* to confirm that Mule can connect with the Cassandra instance.
++
+If the client test is successful, click **OK** to save the configuration. Otherwise, review and correct any invalid parameters and retest.
++
+. Run the Mule app.
```
Comment (kevintroller) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc]:
```suggestion
. Save and run the Mule app.
```

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc [hunk 18]
```diff
@@ -0,0 +1,575 @@
...
+** Update
+** Delete
+* Perform the following operations on an Apache Cassandra table column:
+** Delete the column's value.
+** Rename a column.
+** Add a new column.
+** Drop a column.
+* Execute a Cassandra Query Language (CQL) query.
+* Drop data from the Apache Cassandra database table.
+
```
Comment (kevintroller) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc:316]: Here jumps directly to the XML sample of the application with the operations. I would add an intro line of this XML

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-reference.adoc [hunk 19]
```diff
@@ -0,0 +1,1262 @@
+= Apache Cassandra Connector 4.0 Reference - Mule 4
+:page-aliases: connectors::cassandra/cassandra-connector-reference.adoc
+
+Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#anypoint-connectors[Select]
+
+Anypoint Connector for Cassandra (Cassandra Connector) implements the DataStax Java Driver for Apache Cassandra 3.1.
+
+Release Notes: xref:release-notes::connector/cassandra-connector-release-notes-mule-4.adoc[Cassandra Connector Release Notes]
```
Comment (kevintroller) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-reference.adoc]:
```suggestion

```

Comment (kevintroller) [reply cassandra/4.0/modules/ROOT/pages/cassandra-connector-reference.adoc]: This is in the box of the Connectors Landing page.

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-reference.adoc [hunk 20]
```diff
@@ -0,0 +1,1262 @@
...
+Removes a column
+
+==== Parameters
+[%header,cols="20s,20a,35a,20a,5a"]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | x
+| Table a| String |  The name of the table to use for the operation |  | x
+| Keyspace Name a| String |  The keyspace that contains the table to use |  |
+| Column Name a| String |  The name of the column to removed |  #[payload] |
```
Comment (kevintroller) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-reference.adoc]:
```suggestion
| Column Name a| String |  The name of the column to remove |  #[payload] |
```

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-reference.adoc [hunk 21]
```diff
@@ -0,0 +1,1262 @@
...
+`<cassandra-db:update>`
+
+Executes the update entity operation
+
+==== Parameters
+[%header,cols="20s,20a,35a,20a,5a"]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | x
+| Table a| String |  The name of the table in which the entity is updated |  | x
```
Comment (kevintroller) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-reference.adoc]:
```suggestion
| Table a| String |  The name of the table in which to update the entity |  | x
```

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-studio.adoc [hunk 22]
```diff
@@ -0,0 +1,135 @@
...
+Anypoint Studio (Studio) editors help you design and update your Mule applications, properties, and configuration files.
+
+To add and configure a connector in Studio:
+
+. Create a Mule project.
+. Add the connector to your Mule project.
+. Configure a source for the connector's flow.
+. Add a connector operation to the flow.
+. Configure a global element for the connector.
+. Configure the other connector fields.
```
Comment (kevintroller) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-studio.adoc]:
```suggestion
. <<create-mule-project,Create a Mule project>>.
. <<add-connector-to-project,Add the connector to your Mule project>>.
. <<configure-input-source,Configure a source for the connector's flow>>.
. <<add-connector-operation,Add a connector operation to the flow>>.
. <<configure-global-element,Configure a global element for the connector>>.
. <<configure-other-fields,Configure the other connector fields>>.

```

File: cassandra/4.0/modules/ROOT/pages/cassandra-connector-studio.adoc [hunk 23]
```diff
@@ -0,0 +1,135 @@
...
++
+You can reference a configuration file that contains ANT-style property placeholders (recommended), or you can enter your authorization credentials in the global configuration properties. For information about the benefits of using property placeholders and how to configure them, see xref:connectors::introduction/intro-connector-configuration-overview.adoc[Anypoint Connector Configuration].
++
+. On the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.
+. Optionally, configure a connection that uses Mutual TLS by completing these fields on the Security tab:
++
+[%header,cols="30s,70a"]
+|===
+|Field |Description
+|Endpoint Identification Algorithm | The endpoint identification algorithm used by clients to validate server host name.The default value is an empty string, which means it is disabled. Clients, including client connections created by the broker for inter-broker communication, verify that the broker host name matches the host name in the brokers certificate.
```
Comment (kevintroller) [inline cassandra/4.0/modules/ROOT/pages/cassandra-connector-studio.adoc]:
```suggestion
|Endpoint Identification Algorithm | The endpoint identification algorithm used by clients to validate server host name. The default value is an empty string, which means it is disabled. Clients, including client connections created by the broker for inter-broker communication, verify that the broker host name matches the host name in the brokers certificate.
```

File: cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-conf-task.adoc [hunk 24]
```diff
@@ -1,5 +1,4 @@
 = To Configure Cassandra Authentication
-:page-aliases: 3.9@mule-runtime::cassandra-connector-conf-task.adoc
```
Comment (valkyrie69) [inline cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-conf-task.adoc]: These page aliases are specific to  Mule 3.9 and need to stay (page aliases that are specific to a particular version like in this case should not be removed).

File: cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-create-keyspace-task.adoc [hunk 25]
```diff
@@ -1,5 +1,4 @@
 = To Create a Keyspace
-:page-aliases: 3.9@mule-runtime::cassandra-connector-create-keyspace-task.adoc
```
Comment (valkyrie69) [inline cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-create-keyspace-task.adoc]: Don't delete

File: cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-create-table-task.adoc [hunk 26]
```diff
@@ -1,5 +1,4 @@
 = To Create a Cassandra Table
-:page-aliases: 3.9@mule-runtime::cassandra-connector-create-table-task.adoc
```
Comment (valkyrie69) [inline cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-create-table-task.adoc]: Don't delete

File: cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-datatypes.adoc [hunk 27]
```diff
@@ -1,5 +1,4 @@
 = Cassandra Connector Data Types Reference
-:page-aliases: 3.9@mule-runtime::cassandra-connector-datatypes.adoc
```
Comment (valkyrie69) [inline cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-datatypes.adoc]: Don't delete

File: cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-install-conf-task.adoc [hunk 28]
```diff
@@ -1,5 +1,4 @@
 = To Install and Configure the Cassandra Connector
-:page-aliases: 3.9@mule-runtime::cassandra-connector-install-conf-task.adoc
```
Comment (valkyrie69) [inline cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-install-conf-task.adoc]: Don't delete

File: cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-ops-reference.adoc [hunk 29]
```diff
@@ -1,5 +1,4 @@
 = Cassandra Connector Operations Reference
-:page-aliases: 3.9@mule-runtime::cassandra-connector-ops-reference.adoc
```
Comment (valkyrie69) [inline cassandra/0.3.9/modules/ROOT/pages/cassandra-connector-ops-reference.adoc]: Don't delete

**Review by valkyrie69 — CHANGES_REQUESTED** [review]
Please reinstate the page aliases for the Mule 3.9 topics.

**Review by valkyrie69 — APPROVED** [review]

**Pull request merged**